list_updatable_packages

```diff
@@ -100,9 +136,40 @@ def build_matrix(specs: Iterable[Spec]) -> Generator[dict, None, None]:
                 }
                 yield entry
 
+def build_dependency_matrix(gems: dict, project: str='foreman') -> Generator[dict, None, None]:
+    if DEBUG: print(f'Building dependency matrix for {project}')
+    for gem in gems:
+        try:
+            name = gem['name']
+            path = Path(f'packages/{project}/rubygem-{name}/rubygem-{name}.spec')
+            if path.is_file():
+                spec = Spec(Path(f'packages/{project}/rubygem-{name}/rubygem-{name}.spec'))
+                current_version = spec.version
+            else:
+                continue
+        except subprocess.CalledProcessError as e:  # pylint: disable=invalid-name
+            print('Failed to determine version for', spec.path.as_posix(), str(e), file=sys.stderr)
+        except FileNotFoundError as e:
+            print(f'No path for dependency found: {name}', spec.path.as_posix(), str(e), file=sys.stderr)
+        else:
+            new_version = gem['version']
+            if current_version != new_version:
+                entry = {
+                    'directory': spec.directory,
+                    'package_name': spec.package_name,
+                    'gem_name': spec.gem_name,
+                    'current_version': current_version,
+                    'new_version': new_version,
+                }
+                yield entry
+
 
 def main() -> None:
-    matrix = list(build_matrix(find_specs()))
+    foreman_dependencies = list(build_dependency_matrix(gems=parse_gemfile_lock('foreman'), project='foreman'))
+    katello_dependencies = list(build_dependency_matrix(gems=parse_gemfile_lock('katello'), project='katello'))
+    plugins = list(build_plugin_matrix(find_specs()))
+
+    matrix = plugins + foreman_dependencies + katello_dependencies
```

Comment:
I wonder if you somehow need to do work to detect conflicts between foreman and katello wanting different versions of the same gem where you'd be flip-flopping between them. Just thinking out loud: katello depends on foreman so that would be the super set. Would it be enough (if you look in all project directories using a glob) to only check the Katello lockfile?